plugins/guests/suse/cap/change_host_name.rb

```diff
@@ -5,11 +5,10 @@ class ChangeHostName
         def self.change_host_name(machine, name)
           comm = machine.communicate
 
-          if !comm.test("hostname -f | grep '^#{name}$'", sudo: false)
+          if !comm.test("getent hosts '#{name}'", sudo: false)
             basename = name.split(".", 2)[0]
             comm.sudo <<-EOH.gsub(/^ {14}/, '')
-              echo '#{basename}' > /etc/HOSTNAME
-              hostname '#{basename}'
+              hostnamectl set-hostname '#{basename}'
```

Comment:
Why not `hostnamectl set-hostname '#{name}'`? In the example Vagrantfile that you provide this results in a hostname `leap15-0` instead of `leap15-0.example.com`. Note that if you run `hostnamectl set-hostname leap15-0.example.com` then the bash prompt will still display `vagrant@leap15-0:~>` if that is what you wanted to achieve.
```suggestion
              hostnamectl set-hostname '#{name}'
```

Reply:
But this would write `#{name}` into `/etc/hostname` and `/etc/HOSTNAME`. As far as I know only the host's name, which is the '#{basename}' belongs to these files.

Reply:
That is right, so let's keep it that way.